bears/general/AnnotationBear.py

```diff
@@ -238,9 +242,19 @@ def get_singleline_strings(file,
             A SourceRange object identifying the range of the single-line
             string and the end_position of the string as an integer.
         """
-        end_position = (text.find(string_end, position + 1)
-                        + len(string_end) - 1)
-        newline = text.find("\n", position + 1)
+        try:
+            end_match = next(unescaped_search_for(
+                string_end, text[position + 1:]))
+        except StopIteration:
+            end_match = None
+        end_position = (position + end_match.span()[1] if end_match
+                        else -1)
+        try:
+            newline_match = next(
+                unescaped_search_for("\n", text[position + 1:]))
+        except StopIteration:
+            newline_match = None
+        newline = position + newline_match.span()[1] if newline_match else -1
```

Comment:
they can't be right? they'll raise error at different times and different behavior is expected at those times